src/Chirp.Core/Classes/CheepRepository.cs

```diff
@@ -60,6 +60,24 @@ public int GetCheepCountByName(string authorName)
             .Include(c => c.Author)
             .Count(c => EF.Functions.Collate(c.Author.UserName, "NOCASE") == authorName);
     }
+    
+    public int GetCheepCountByAuthors(List<Author> followedAuthors, string currentUserId)
+    {
+        // Ensure that followedAuthors is not null to avoid NullReferenceException
+        followedAuthors ??= new List<Author>();
+
+        // Add the current user's ID to the list of author IDs
+        var authorIds = followedAuthors.Select(a => a.Id).ToList();
+        if (!authorIds.Contains(currentUserId))
+        {
+            authorIds.Add(currentUserId);
+        }
+
+        // Return the count of cheeps where the author ID is in the list of author IDs
```

Comment:
Should probably remove these unnecessary comments. The method is self descriptive